sopel/module.py

```diff
@@ -1,1109 +1,11 @@
-            self.result = result
-        elif result is not None:
-            self.result = [result]
-        else:
-            self.result = None
-        self.use_re = re
-        self.msg = msg
-        self.privmsg = privmsg
-        self.admin = admin
-        self.owner = owner
-        self.repeat = repeat
-        self.online = online
-        self.vcr = vcr
-
-        if isinstance(ignore, list):
-            self.ignore = ignore
-        elif ignore is not None:
-            self.ignore = [ignore]
-        else:
-            self.ignore = []
-
-        self.user_help = user_help
-
-    def __call__(self, func):
-        if not hasattr(func, "example"):
-            func.example = []
-
-        import sys
-
-        import sopel.test_tools  # TODO: fix circular import with sopel.bot and sopel.test_tools
-
-        # only inject test-related stuff if we're running tests
-        # see https://stackoverflow.com/a/44595269/5991
-        if 'pytest' in sys.modules and self.result:
-            # avoids doing `import pytest` and causing errors when
-            # dev-dependencies aren't installed
-            pytest = sys.modules['pytest']
-
-            test = sopel.test_tools.get_example_test(
-                func, self.msg, self.result, self.privmsg, self.admin,
-                self.owner, self.repeat, self.use_re, self.ignore
-            )
-
-            if self.online:
-                test = pytest.mark.online(test)
-
-            if self.vcr:
-                test = pytest.mark.vcr(test)
-
-            sopel.test_tools.insert_into_module(
-                test, func.__module__, func.__name__, 'test_example'
-            )
-            sopel.test_tools.insert_into_module(
-                sopel.test_tools.get_disable_setup(), func.__module__, func.__name__, 'disable_setup'
-            )
-
-        record = {
-            "example": self.msg,
-            "result": self.result,
-            # flags
-            "is_private_message": self.privmsg,
-            "is_help": self.user_help,
-            "is_pattern": self.use_re,
-            "is_admin": self.admin,
-            "is_owner": self.owner,
-            # old-style flags
-            # TODO: to be removed in Sopel 8.0
-            "privmsg": self.privmsg,
-            "admin": self.admin,
-            "help": self.user_help,
-        }
-        func.example.append(record)
-        return func
-
-
-def output_prefix(prefix):
-    """Decorate a function to add a prefix on its output.
-
-    :param str prefix: the prefix to add (must include trailing whitespace if
-                       desired; Sopel does not assume it should add anything)
-
-    Prefix will be added to text sent through:
-
-    * :meth:`bot.say <sopel.bot.SopelWrapper.say>`
-    * :meth:`bot.notice <sopel.bot.SopelWrapper.notice>`
-
-    """
-    def add_attribute(function):
-        function.output_prefix = prefix
-        return function
-    return add_attribute
+from sopel.plugin import *  # noqa
```

Comment:
I'd use the old `__all__` list, so we don't export any names in `module` that weren't present in 7.0.x. Anything new should be _forced_ to import from `plugin` instead.

Reply:
Oh, that's fair, yes.